### PR TITLE
Blog twitter cards 1169

### PR DESCRIPTION
**Pull request by astephanus**
### Description
Implements Twitter Card meta data by both correcting, and completing the previously incorrect, and incomplete implementation. 

- Correctly accounts for field length restrictions. 
- Allows for various ways to define `twitter:description`; either explictly using `twittercard.description` Front Matter with a fallback to `excerpt`, and then to `meta_description`. 
- Allows for specification of `twitter:creator` with a fallback to the first author (The fallback should to be discussed.)
- Allows for explicit specification of `twitter:site` using a Front Matter property named `twittercard.site` with a default of `@OpenSearchProj`.
- Template logic sets the Twitter Card type to `summary_large_image` for content with an image and `summary` for content without an image.
- Template logic will use `twittercard.image` if defined, or `feature_image` if not defined. 
- Adds `twittter:image:alt` for accessibility support by using `twittercard.image_alt` in the Front Matterr.
- Removes a default image from the `head.html` template in accordance with Twitter's best practices about not using the same image for different content.
- Adds values for `twittercard.description` taken from first paragraph page content for blog posts for posts that did not already have a value defined for `excerpt`. Markdown code has been stripped from those values, and some have been  subjectively truncated based on the fact that Twitter allows only for 200 characters across 3 lines (Only for content that was obviously very, very much longer). Values are still explicitly truncated in the `head.html` template as not to violate Twitter's rules.

Solution requires validation through Twitter's Card Validator (https://cards-dev.twitter.com/validator). It is not of this moment certain how to do so from localhost. 

Values that have been added for `twittercard.description` should be reviewed by editorial staff.
 
### Issues Resolved
https://github.com/opensearch-project/project-website/issues/1169

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

**Comment by nateynateynate** [comment]
I'll check this out best I can, but we might need to stage this and then run some twitter card previewers before we hit production just for a few sets of eyes. 

@krisfreedain : Mind working with me after I put eyeballs on this?

File: _includes/head.html [hunk 1]
```diff
@@ -20,31 +20,42 @@
...
-      {% assign author_short_name = page.authors | first  %}
-      {% assign author = site.authors | where: 'short_name', author_short_name | first %}
-      {% if author.twitter %}
-        <meta name="twitter:site" content="@{{ author.twitter }}" />
-      {% endif %} 
+    {% if page.twittercard.image or page.feature_image %}
+    <meta name="twitter:card" content="summary_large_image">
+    {% else %}
+    <meta name="twitter:card" content="summary">
     {% endif %}
```
Comment (nateynateynate) [inline _includes/head.html:28]: I believe that we can also specify `player` for the meta keyword `twitter:card` - we may want to some day tweet something that plays automagically for people on twitter.

Comment (astephanus) [reply _includes/head.html:28]: @nateynateynate Would a reasonable solution be to accept a specific card type indicated in the Front Matter and then fall back accordingly if not specified?

Comment (astephanus) [reply _includes/head.html:30]: @nateynateynate I have updated the `head.html` template with the player type logic.

**Review by nateynateynate — APPROVED** [review]
LEts rawk.

**Pull request merged**